libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -161,6 +165,20 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishDRepPulser (nes ^. newEpochStateGovStateL . drepPulsingStateGovStateL)
 
+-- | Query pool stake distribution.
+querySPOStakeDistr ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  Set (KeyHash 'StakePool (EraCrypto era)) ->
+  -- | Specify pool key hashes whose stake distribution should be returned. When this set is
+  -- empty, distributions for all of the pools will be returned.
+  (Map (KeyHash 'StakePool (EraCrypto era)) (IndividualPoolStake (EraCrypto era)))
```

Comment:
For now, I added the whole `IndividualPoolStake`. Would it be better to only track the `individualTotalPoolStake` field?

Reply:
It would be better to track individual `Coin`, especially since `individualPoolStake` is not updated (nor used) by the pulser. We have plans on fixing that, but there is no immediate pressure on doing it.